src/library_pthread.js

```diff
@@ -98,10 +98,20 @@ var LibraryPThread = {
     },
     initMainThread() {
 #if PTHREAD_POOL_SIZE
-      var pthreadPoolSize = {{{ PTHREAD_POOL_SIZE }}};
-      // Start loading up the Worker pool, if requested.
-      while (pthreadPoolSize--) {
-        PThread.allocateUnusedWorker();
+      preallocateWorkers = () => {
+        var pthreadPoolSize = {{{ PTHREAD_POOL_SIZE }}};
+        // Start loading up the Worker pool, if requested.
+        while (pthreadPoolSize--) {
+         PThread.allocateUnusedWorker();
+        }
+      };
+#if expectToReceiveOnModule('mainScriptUrlOrBlobPromise')
```

Comment:
What is `mainScriptUrlOrBlobPromise`?   I don't see that anywhere in the existing source code?  Is the idea that someone could add this to `-sINCOMING_JS_MODULE_API`?    If so I wonder if we should document it somewhere.

If that creation of the main thread needs to be delayed until the `mainScriptUrlOrBlob` is ready can't this be done via existing mechanisms?  For example, any of these approaches:

1. Using MODULARIZE and delaying the instantiation of the module.
2. Delaying the import of the emscripten-generated script
3. Using `addRunDependency` ?  (this one might not work since I'm not sure it blocks `initMainThread`.